students/student_framllat/session08/test_the_circle.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+
+from the_circle import Circle, Sphere
+from math import pi
+import random
+
+def test_init():
+    Circle(10)
+
+def test_radius():
+    c = Circle(25)
+
+    assert c.radius == 25
+
+def test_diameter():
+    c = Circle(5)
+    assert c.diameter == 10
+
+def test_set_radius():
+    c = Circle(8)
+    c.diameter = 16
+
+    assert c.radius == 8
+    assert c.diameter == 16
+
+
+def test_area():
+    c = Circle(2)
+
+    assert c.area == pi * 4
+
+def test_add_circle():
+    c1 = Circle(12)
+    c2 = Circle(7)
+    c3 = c1 + c2
+
+    assert c3.radius == 19
+
+def test_mult_circle():
+    c1 = Circle(4)
+    c2 = c1 * 4
+    
+    assert c2.radius == 16
+
+def test_str():
+    c = Circle(10)
+
+    assert str(c) == 'Circle with radius: 10.000000'
+
+def test_equal():
+    c8 = Circle(8)
+    c9 = Circle(8.0)
+
+    assert c8 == c9
+    assert c8 <= c9
+    assert c8 >= c9
+
+def test_less():
+    c1 = Circle(2)
+    c2 = Circle(3)
+
+    assert c1 < c2
+    assert c1 <= c2
+
+def test_sort():
+    a_list = [Circle(20), Circle(10), Circle(15), Circle(5)]
+
+    a_list.sort()
+
+    assert a_list[0] == Circle(5)
+    assert a_list[3] == Circle(20)
+    assert a_list[0] < a_list[1] < a_list[2] < a_list[3]
+
+def test_sphere_vol():
+    s = Sphere(4)
+
+    print(s.volume())
+    assert s.volume() == 268.082573106329
```

Comment:
notice that you have to call `.volume`, but not `.area`, etc... that's because it's not a property.